packages/esm-patient-registration-app/src/offline.resources.ts

```diff
@@ -75,17 +75,21 @@ export async function fetchPatientIdentifierTypesWithSources(): Promise<Array<Pa
   // @ts-ignore Reason: The required props of the type are generated below.
   const identifierTypes: Array<PatientIdentifierType> = patientIdentifierTypes.filter(Boolean);
 
-  const [autoGenOptions, ...allIdentifierSources] = await Promise.all([
+  const [autoGenOptions, identifierSourcesResponse] = await Promise.all([
     fetchAutoGenerationOptions(),
-    ...identifierTypes.map((identifierType) => fetchIdentifierSources(identifierType.uuid)),
+    fetchIdentifierSources(identifierTypes.map((identifierType) => identifierType.uuid)),
   ]);
 
+  const allIdentifierSources = identifierSourcesResponse.data.results;
+
   for (let i = 0; i < identifierTypes?.length; i++) {
-    identifierTypes[i].identifierSources = allIdentifierSources[i].data.results.map((source) => {
-      const option = find(autoGenOptions.data.results, { source: { uuid: source.uuid } });
-      source.autoGenerationOption = option;
-      return source;
-    });
+    identifierTypes[i].identifierSources = allIdentifierSources
+      .filter((source) => source.identifierTypeUuid === identifierTypes[i].uuid)
```

Comment:
AFAICT, the source doesn't have an identifierTypeUuid. I think that's ok, we just need the map, not the filter.

Reply:
@ibacher Thanks so much! let me do a fix.